### DISCUSSION TITLE
How many objects can Yolo v8 track?

### DISCUSSION THREAD

**Discussion by tonyreina** [ultralytics/ultralytics#12997] [General]
I'd like to track the movement of potentially thousands of cells in a fluid using a microscope. 

Is there an upper limit on how many objects the Yolo v8 model can track simultaneously? I realize I could potentially scale things by dividing the image into tiles and doing inference on each tile before stitching back together, but wanted to get a sense of how many detections can be made per tile.

Thanks.
-Tony

**Comment by glenn-jocher** [comment]
@tonyreina hi Tony,

YOLOv8 can handle a large number of objects in a single frame, but the practical limit depends on the complexity of the scene and the hardware capabilities. For tracking thousands of cells, your approach of tiling the image and processing each tile separately is indeed effective. This method helps manage the load and can improve tracking accuracy and performance.

There isn't a hardcoded limit on detections per tile, but keep in mind that performance may vary based on the resolution of each tile and the computational resources available. If you encounter performance issues, consider adjusting the tile size or the model's input resolution.

Here's a basic example of how you might set up your model for tracking:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8n.pt')

# Process each tile
results = model.track('tile.jpg')
```

Adjust the tiling strategy and model configuration to best fit your setup. Good luck with your project!

**Reply by tonyreina** [reply]
Many thanks Glenn!

**Reply by glenn-jocher** [reply]
You're welcome! If you have any more questions or need further assistance, feel free to ask. Happy coding! 😊
